src/main/java/org/openstreetmap/atlas/generator/creator/RawAtlasCreator.java

```diff
@@ -0,0 +1,375 @@
+        final Sharding sharding = AtlasSharding.forString(shardingName, Maps.stringMap());
+        final Sharding pbfSharding = pbfShardingName != null
+                ? AtlasSharding.forString(shardingName, Maps.stringMap()) : sharding;
+        final String countryName = (String) command.get(COUNTRY);
+        final File output = (File) command.get(OUTPUT);
+        final RawAtlasFlavor atlasFlavor = (RawAtlasFlavor) command.get(ATLAS_FLAVOR);
+        final boolean useJavaFormat = (boolean) command.get(USE_JAVA_ATLAS);
+
+        PbfLoader.setAtlasSaveFolder(output);
+        final PackedAtlas atlas = runGenerationForFlavor(atlasFlavor, countryBoundaryMap,
+                shardToBuild, pbfPath, slicedCachePath, failFastOnSlicedCacheMiss, pbfScheme,
+                pbfSharding, sharding, countryName);
+        if (useJavaFormat)
+        {
+            atlas.setSaveSerializationFormat(AtlasSerializationFormat.JAVA);
+        }
+        else
+        {
+            atlas.setSaveSerializationFormat(AtlasSerializationFormat.PROTOBUF);
+        }
+        atlas.save(output.child(countryName + CountryShard.COUNTRY_SHARD_SEPARATOR
+                + shardToBuild.getName() + FileSuffix.ATLAS));
+        atlas.saveAsGeoJson(output.child(countryName + "_" + shardToBuild.getName() + ".geojson"));
+
+        return 0;
+    }
+
+    @Override
+    protected SwitchList switches()
+    {
+        return new SwitchList().with(BOUNDARIES, TILE, SHARDING_TYPE, PBF_PATH, SLICED_CACHE_PATH,
+                FAIL_FAST_CACHE_MISS, PBF_SHARDING, COUNTRY, OUTPUT, ATLAS_FLAVOR, USE_JAVA_ATLAS);
+    }
+
+    private Function<Shard, Optional<Atlas>> atlasFetcher(final String countryName,
+            final String slicedCachePath, final boolean failFastOnSlicedCacheMiss,
+            final String pbfPath, final CountryBoundaryMap countryBoundaryMap)
+    {
+        return (Function<Shard, Optional<Atlas>> & Serializable) shard ->
+        {
+            final String shardName = shard.getName();
+            final String atlasName = countryName + "_" + shardName + FileSuffix.ATLAS;
+            final AtlasResourceLoader loader = new AtlasResourceLoader();
+            final Path slicedCacheFile = Paths.get(slicedCachePath, atlasName);
+            final RawAtlasCreator creator = new RawAtlasCreator();
+
+            Optional<Atlas> atlasOption = Optional
+                    .ofNullable(loader.load(new File(slicedCacheFile.toString())));
+
+            if (!atlasOption.isPresent())
+            {
+                logger.warn("Sliced cache miss for {}, generating...", atlasName);
+
+                try
+                {
+                    atlasOption = Optional.ofNullable(creator.generateSlicedAtlasFromScratch(
+                            pbfPath, countryName, countryBoundaryMap, shard));
+                }
+                catch (final Exception exception)
+                {
+                    atlasOption = Optional.empty();
```

Comment:
We should probably log the exception here - would assume if there is a valid PBF path and we fail to generate the sliced atlas file for it - we have an issue.

Reply:
Done.